spec/yaml/swiftnav/sbp/skylark.yaml

```diff
@@ -0,0 +1,50 @@
+# Copyright (C) 2018-2022 Swift Navigation Inc.
+# Contact: https://support.swiftnav.com
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+package: swiftnav.sbp.skylark
+description: Messages from the Skylark corrections service.
+stable: False
+public: True
+include:
+  - types.yaml
+definitions:
+  - MSG_ACKNOWLEDGE:
+      id: 0x0FA0
+      short_desc: Acknowledgement message in response to a request for corrections
+      fields:
+        - request_id:
+            type: u8
+            desc: Echo of the request ID field from the corresponding CRA message, or 255 if no request ID was provided.
+        - area_id:
+            type: u32
+            desc: Echo of the Area ID field from the corresponding CRA message.
+        - response_code:
+            type: u8
+            desc: Reported status of the request.
+            fields:
+              - 0-7:
+                  desc: Response code
+                  values:
+                      - 0: Ok
+                      - 1: Out of coverage
+                      - 2: Forbidden
+                      - 3: Invalid request
+                      - 4: Invalid area id
```

Comment:
this generates "masking" code:

```c
#define SBP_ACKNOWLEDGE_RESPONSE_CODE_MASK (0xffu)
#define SBP_ACKNOWLEDGE_RESPONSE_CODE_SHIFT (0u)
#define SBP_ACKNOWLEDGE_RESPONSE_CODE_GET(flags)               \
  ((u8)((u8)((flags) >> SBP_ACKNOWLEDGE_RESPONSE_CODE_SHIFT) & \
        SBP_ACKNOWLEDGE_RESPONSE_CODE_MASK))
#define SBP_ACKNOWLEDGE_RESPONSE_CODE_SET(flags, val)                      \
  do {                                                                     \
    (flags) = (u8)((flags & (~(SBP_ACKNOWLEDGE_RESPONSE_CODE_MASK          \
                               << SBP_ACKNOWLEDGE_RESPONSE_CODE_SHIFT))) | \
                   (((val) & (SBP_ACKNOWLEDGE_RESPONSE_CODE_MASK))         \
                    << (SBP_ACKNOWLEDGE_RESPONSE_CODE_SHIFT)));            \
  } while (0)
```

Which is strange as this variable doesn't represent a bit mask field. Is there any type that represents enumerator types rather bit fields? 

With the current setup, we are going to be limited to only being able to add 3 more error codes in the future rather than 200 more possible values.

Reply:
Looks like that generates similar code. I guess if you have 

```yaml
fields:
   - 0-7: ... # or 0 to whatever size int it is
```

we could probably just skip generating the mask/shifting bit and just generate the constants like - https://github.com/swift-nav/libsbp/blob/0e72dc09788f2b8dbf92e62f4ae23373bc39c954/c/include/libsbp/gnss_macros.h#L32